spec/services/hmrc/interface/result_service_spec.rb

```diff
@@ -0,0 +1,163 @@
+      )
+  end
+
+  describe '#call' do
+    subject(:call) { described_class.call(hmrc_response) }
+
+    it { is_expected.to eql expected_body }
+  end
+
+  describe '.call' do
+    subject(:call) { submission_result.call }
+
+    context 'when the submission has completed successfully' do
+      it { is_expected.to eql expected_body }
+    end
+
+    context 'when the submission is still in progress' do
+      let(:expected_status) { 202 }
+      let(:expected_body) do
+        {
+          submission: hmrc_response.submission_id,
+          status: 'in_progress',
+          _links: [
+            {
+              href: "https://main-laa-hmrc-interface-uat.apps.live-1.cloud-platform.service.justice.gov.uk/api/v1/submission/status/#{hmrc_response.submission_id}"
+            }
+          ]
+        }
+      end
+
+      it 'does not raise an error' do
+        expect { call }.not_to raise_error
+      end
+    end
+
+    context 'when the submission has failed' do
+      let(:expected_status) { 202 } # TODO: Response should be 200, also json response should equal - AP-2639 addresses
+      # {
+      #   data: [
+      #     { correlation_id: hmrc_response.submission_id },
+      #     { error: 'submitted client details could not be found in HMRC service' }
+      #   ]
+      # }
+      let(:expected_body) do
+        {
+          submission: hmrc_response.submission_id,
+          status: 'failed',
+          _links: [
+            {
+              href: "https://main-laa-hmrc-interface-uat.apps.live-1.cloud-platform.service.justice.gov.uk/api/v1/submission/status/#{hmrc_response.submission_id}"
+            }
+          ]
+        }
+      end
+
+      it 'does not raise an error' do
+        expect { call }.not_to raise_error
+      end
+
+      it { is_expected.to eql expected_body }
+    end
+
+    context 'when an error occurs in the result process' do
+      let(:get_url) { %r{http.*laa-hmrc-interface.*apps.live-1.cloud-platform.service.justice.gov.uk/api/v1/submission/result/.*} }
+      before do
+        stub_request(:get, get_url)
+          .to_raise(StandardError)
+      end
+
+      context 'http status 422' do
+        before do
+          stub_request(:get, get_url)
+            .to_return(body: error_response, status: 422)
+        end
+
+        it 'raises an exception' do
+          expect { call }.to raise_error HMRC::SubmissionError, /Couldn't find ProceedingType/
+        end
+      end
+
+      context 'other failing http status' do
+        before do
+          stub_request(:get, get_url)
+            .to_return(body: '', status: 503)
+        end
+
+        it 'raises an exception' do
+          expect {
+            described_class.call(hmrc_response)
+          }.to raise_error HMRC::SubmissionError, /Bad Request: URL:/
+        end
+      end
+
+      context 'connection error' do
+        it 'creates a LegalFramework::Submission error and writes a history record with a backtrace' do
```

Comment:
I think this is a C&P error 😁

Reply:
How do you feel about this being named `SubmissionError`? As it became used in submission _and_ result, would `HMRC::InterfaceError` be better?

Reply:
I deleted these and refactored the name anyway!